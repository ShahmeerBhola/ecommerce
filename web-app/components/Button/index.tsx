import React, { FC } from 'react';
import Link from 'next/link';

type OwnProps = {
    inverted?: boolean;
    textSize?: Styles.FontSizes;
    paddingX?: string;
    link?: string;
};

export type Props = OwnProps &
    React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>;

const Button: FC<Props> = ({
    children,
    link,
    className,
    inverted = false,
    textSize = 'text-sm',
    paddingX = 'px-5',
    type = 'button',
    ...rest
}) => {
    let extraClassNames: string;
    if (inverted) {
        extraClassNames =
            'text-red-100 hover:text-red-200 bg-transparent border-solid border-2 border-red-100 hover:border-red-200';
    } else {
        extraClassNames = 'text-white bg-red-100 hover:bg-red-200';
    }

    let classes = `${textSize} font-primary font-bold py-2 ${paddingX} uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none`;
    classes += ` ${className}`;
    classes += ` ${extraClassNames}`;

    const Butt = (): JSX.Element => (
        <button className={classes} type={type} {...rest}>
            {children}
        </button>
    );

    if (link) {
        return (
            <Link href={link}>
                <a>
                    <Butt />
                </a>
            </Link>
        );
    }

    return <Butt />;
};

export default Button;
