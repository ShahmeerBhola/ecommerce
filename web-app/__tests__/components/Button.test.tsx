import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('Button component', () => {
  test('it renders properly', () => {
    const tree = renderer.create(<Button>Click</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with inverted prop set to true', () => {
    const tree = renderer.create(<Button inverted>Click</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when passed the link prop', () => {
    const tree = renderer.create(<Button link="https://joeyorlando.ca">Click</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when passed the type prop', () => {
    const tree = renderer.create(<Button type="submit">Submit</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
