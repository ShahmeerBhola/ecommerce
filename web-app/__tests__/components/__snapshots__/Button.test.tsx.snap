// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component it renders properly 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none text-base text-white bg-red-100 hover:bg-red-200"
  type="button"
>
  Click
</button>
`;

exports[`Button component it renders properly when passed the link prop 1`] = `
<a
  href="https://joeyorlando.ca"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <button
    className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none text-base text-white bg-red-100 hover:bg-red-200"
    type="button"
  >
    Click
  </button>
</a>
`;

exports[`Button component it renders properly when passed the type prop 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none text-base text-white bg-red-100 hover:bg-red-200"
  type="submit"
>
  Submit
</button>
`;

exports[`Button component it renders properly with inverted prop set to true 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none text-base text-red-100 hover:text-red-200 bg-transparent border-solid border-2 border-red-100 hover:border-red-200"
  type="button"
>
  Click
</button>
`;
